app/renderer/js/components/context-menu.ts

```diff
@@ -0,0 +1,112 @@
+import {remote} from 'electron';
+import WebView from './webview';
+import handleExternalLink from './handle-external-link';
+import * as t from '../utils/translation-util';
+const {clipboard, Menu, MenuItem} = remote;
+
+export const contextMenu = (webview: WebView, event: any) => {
+	const webContents = webview.$el.getWebContents();
+	const props = event.params;
+	const isText = Boolean(props.selectionText.length);
+	const isLink = Boolean(props.linkURL);
+
+	const getContextMenu = (): Electron.MenuItemConstructorOptions[] => {
+		return [{
+			label: t.__('Add to Dictionary'),
+			visible: props.isEditable && isText && props.misspelledWord,
+			click(_item) {
+				webContents.session.addWordToSpellCheckerDictionary(props.misspelledWord);
+			}
+		}, {
+			type: 'separator'
```

Comment:
bug: Separator visible on ubuntu even if word is not misspelled.
`visible` attribute doesn't seem to work here too.